eduNEXT/edx-platform
# [WIP] CAM-10 - Generating reports

# Description

This PR builds three types of reports: by section and by assignment type and by problem.

## How to test

Go to courses/your_course_id/instructor#view-data_download and you'll see three new buttons at the bottom. Just generate these reports normally.

## Screenshots

### Report by section

![by_section](https://user-images.githubusercontent.com/3374009/40257976-b259a95a-5ab5-11e8-9fe4-7c0a354ce012.png)

### Report by assignment type

![assignment_type](https://user-images.githubusercontent.com/3374009/40257961-a75021ba-5ab5-11e8-9cfe-fbb8e37489a4.png)

### Report by problem

Screenshots are splited for a better visualization

![captura de pantalla de 2018-05-18 16-11-10](https://user-images.githubusercontent.com/3374009/40258132-2e8e87d4-5ab6-11e8-8ee8-75a9a14953a9.png)

![captura de pantalla de 2018-05-18 16-12-38](https://user-images.githubusercontent.com/3374009/40258183-5af18146-5ab6-11e8-8af7-4397640613d6.png)

![captura de pantalla de 2018-05-18 16-13-37](https://user-images.githubusercontent.com/3374009/40258222-7a70598e-5ab6-11e8-934e-b01fe0954d75.png)

## Course structure

This course structure was thought to cover most of the cases in a course to ensure the CSV's output. 

- Section 1
    - Subsection One (Homework)
        - Unit
            - Dropdown problem
    - Subsection 2  (Lab)
        - Unit 
            - Dropdown problem
    - Subsection 3 (Homework)
        - Unit
            - Checkboxes problem

- Section 2
    - Subsection 4 (Homework)
        - Unit
            - Checkboxes problem
    - Subsection 5 (Lab)
        - Unit
            -  Checkboxes problem
   - Subsection 6 (Midterm Exam)
        - Unit
            - Dropdown problem
- Section 3
    - Subsection (Midterm Exam)
         - Unit
            - Multiple Choice problem

**Grading Policy:**
HomeWork -> Weight: 15, Total number: 3, Droppable: 2
Lab ->  Weight: 15, Total number: 2, Droppable: 1
Midterm Exam ->  Weight: 70, Total number: 2, Droppable: 0

## Reviewers
